### PR TITLE
53 evict users

### PR DESCRIPTION
**Pull request by DL-USAF**
image

### CODE REVIEW

File: src/datawave_cli/authorizations_interactions.py [hunk 1]
```diff
@@ -28,24 +29,51 @@ def authorization_whoami(self):
...
+        return ai.authorization_whoami()
+    elif args.evict_users:
+        return ai.authorization_evict_users()
+    else:
+        log.warning("No cmd given, exiting without running anything. Pass '--whoami' or '--evict_users'.")
 
 
 @click.command
 @common_options
+@click.option("--whoami", is_flag=True, help="Flag to denote to run the whoami endpoint.")
```
Comment (JTR-USAF) [inline src/datawave_cli/authorizations_interactions.py:75]: Should we make one of these the default option? Probably whoami cause that is going to be called a lot more often than evict

File: src/tests/test_authorization_interactions.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+
+def test_whoami(mocker, authorization_interactions, mock_log_http_response, test_user):
+    """Tests the whoami function call to validate we are getting correct responses
+    and performing correct logic."""
+    mock_requests_get = mocker.patch('requests.get')
+    mock_response = mocker.Mock()
+    mock_response.status_code = 200
+    mock_response.json.return_value = test_user
+    mock_requests_get.return_value = mock_response
```
Comment (JTR-USAF) [inline src/tests/test_authorization_interactions.py:58]: Why aren't you using your new utils fixture?

File: src/tests/test_authorization_interactions.py
Comment (JTR-USAF) [inline src/tests/test_authorization_interactions.py:1]: image

File: src/tests/test_authorization_interactions.py (hunk 2 above)
Comment (DL-USAF) [reply src/tests/test_authorization_interactions.py:58]: the function was written after...

File: src/tests/test_authorization_interactions.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        (500, HTTPError("500 Server Error: Internal Server Error for url")),
+        (None, Timeout("The request timed out"))
+    ],
+    ids=[
+        'Forbidden',
+        'Internal Server Error',
+        'Timeout',
+    ]
+)
+def test_whoami_json_errors(mock_status_code, mock_exception, mocker, authorization_interactions):
```
Comment (JTR-USAF) [inline src/tests/test_authorization_interactions.py:87]: Not testing JSON errors. Inf act you're not testing JSON errors at all!

File: src/tests/test_authorization_interactions.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        cert=authorization_interactions.cert,
+        headers=authorization_interactions.headers,
+        verify=False
+    )
+
+    mock_log_http_response.assert_called_once_with(mock_response, authorization_interactions.log)
+    authorization_interactions.log.info.assert_any_call("Requesting all users to be evicted from DW...")
+
+    assert resp == mock_response
+    assert resp.text is "all entries evicted"
```
Comment (JTR-USAF) [inline src/tests/test_authorization_interactions.py:119]: What did we say about `is` vs `==`. Bad david. Be better.

**Review by NC-USAF — APPROVED** [review]

**Review by JTR-USAF — APPROVED** [review]

**Pull request merged**